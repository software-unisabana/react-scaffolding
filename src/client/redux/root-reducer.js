import { combineReducers } from 'redux';

import itemReducer from './item/item.reducer';

const rootReducer = combineReducers({
	products: itemReducer,
});

export default rootReducer;
