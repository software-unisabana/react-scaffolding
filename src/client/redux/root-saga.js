import { all, call } from 'redux-saga/effects';

import { itemSagas } from './item/item.sagas';

export default function* rootSaga() {
	yield all([call(itemSagas)]);
}
