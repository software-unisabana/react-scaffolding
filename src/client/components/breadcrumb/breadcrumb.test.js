import React from 'react';
import { mount } from 'enzyme';
import { combineReducers, createStore } from 'redux';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';

import BreadCrumb from './breadcrumb.component';

export const createMockStore = ({ state, reducers }) => {
	const store = createStore(combineReducers(reducers), state);
	return {
		...store,
	};
};

describe('BreadCrumb Component', () => {
	let wrapper;
	let store;

	beforeEach(() => {
		const mockReducer = (state = { products: { categories: [] } }, action) =>
			state;
		store = createMockStore({
			state: { products: { categories: ['Auido', 'Audio y video'] } },
			reducers: { products: mockReducer },
		});
		wrapper = mount(
			<BrowserRouter>
				<Provider store={store}>
					<BreadCrumb />
				</Provider>
			</BrowserRouter>
		);
	});
	it('should render BreadCrumb component', () => {
		expect(wrapper).toMatchSnapshot();
	});
	it('should render an equal number of Sections as the Categories', () => {
		expect(wrapper.find("[className='category']").length).toEqual(2);
	});
});
