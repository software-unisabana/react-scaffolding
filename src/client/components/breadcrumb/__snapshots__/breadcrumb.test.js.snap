// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BreadCrumb Component should render BreadCrumb component 1`] = `ReactWrapper {}`;
